requests/models.py

```diff
@@ -420,8 +420,11 @@ def prepare_body(self, data, files, json=None):
         length = None
 
         if not data and json is not None:
-            content_type = 'application/json'
-            body = complexjson.dumps(json)
+            # When urllib3 uses pyOpenSSL, it can only resume large uploads
+            # properly if receiving a bytes-like object. In Python 2, json.dumps()
+            # returns just that, but Python 3 returns a Unicode string.
+            content_type = 'application/json; charset=utf-8'
```

Comment:
Heh, little-known fact: the content-type for JSON (`application/json`) does not take a `charset` parameter, and anyone who sets it is doing it wrong. Mind removing the charset parameter?